src/components/GameBoard.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import { Square, Row } from '../constants';
+import Game from '../tic-tac-toe';
+
+class GameBoard extends React.Component<any, any> {
+  constructor(props: {}) {
+    super(props);
+    this.state = {
+      board: [
+        [undefined, undefined, undefined],
+        [undefined, undefined, undefined],
+        [undefined, undefined, undefined],
+      ],
+      boardSize: 0,
+      gameStatus: null,
+    };
+  }
+
+  public handleChange = (event: React.FormEvent<HTMLInputElement>): void => {
+    this.setState({
+      boardSize: event.currentTarget.value,
```

Comment:
My thought was to have the form update state as the user types so that I can validate the input in real-time, and then when they hit submit, it creates the game and displays the board. Unless I'm misunderstanding, if I set both the board and board size every time the input changes, wouldn't I be creating a new game multiple times?

I'm thinking of moving this logic into a parent component called `Game`, which handles the game creation and then sends the board size to a `Board` component which just draws a board based on the given size and handles user input.

Reply:
I think that the distinction is between "setting a board size" and "starting a game." I'm thinking of it like this:

```
user types ->
  boardSize is set

board ->
  new Board(this.state.boardSize)

user submits -> 
  new Game(board: board())
```

Where the components are flattened out to work with only their direct dependencies.  Given the above, it might be cleaner to separate the Game component from the form to create one.